shared/actions/login/index.js

```diff
@@ -475,6 +479,29 @@ function makeKex2IncomingMap (dispatch, getState) : incomingCallMapType {
       generateQRCode(dispatch, getState)
       dispatch(askForCodePage(phrase => { response.result({phrase, secret: null}) }, response))
     },
+    'keybase.1.provisionUi.chooseDeviceType': ({sessionID, kind}, response) => {
+      const store = getState().login.codePage
+      appendRouteElement((
+        <ExistingDevice
+          myDeviceRole={store.myDeviceRole}
+          onSubmit={kind => {
+            let deviceType: DeviceType
+            switch (kind) {
```

Comment:
could also do a data driven approach since its just pull out one var, but its fine
```
const deviceType: ?DeviceType = {
[Constants.codePageDeviceRoleNewComputer]: enums.provisionUi.DeviceType.desktop,
[Constants.codePageDeviceRoleNewPhone]: enums.provisionUi.DeviceType.mobile
}[kind]
```